src/test/java/com/the/dark/side/crew/fejsbuk/mapper/LikeMapperTest.java

```diff
@@ -0,0 +1,114 @@
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class LikeMapperTest {
+
+    @InjectMocks
+    private LikeMapper likeMapper;
+
+    @Mock
+    private UserRepository userRepository;
+
+    @Mock
+    private PostRepository postRepository;
+
+    @BeforeEach
+    void init() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    void whenToDtoThenReturnLikeDto() {
+        LikeEntity likeEntity = mock(LikeEntity.class);
+        PostEntity postEntity = mock(PostEntity.class);
+        UserEntity userEntity = mock(UserEntity.class);
+        LocalDateTime createdAt = LocalDateTime.now();
+        long likeId = 3L;
+        long postId = 1L;
+        long userId = 2L;
+
+        when(likeEntity.getId()).thenReturn(likeId);
+        when(likeEntity.getCreatedAt()).thenReturn(createdAt);
+        when(likeEntity.getUserEntity()).thenReturn(userEntity);
+        when(likeEntity.getPostEntity()).thenReturn(postEntity);
+        when(userEntity.getId()).thenReturn(userId);
+        when(postEntity.getId()).thenReturn(postId);
+
+        LikeDto result = likeMapper.toDto(likeEntity);
+        assertEquals(likeId, result.getId());
+        assertEquals(createdAt, result.getCreatedAt());
+        assertEquals(postId, result.getPostId());
+        assertEquals(userId, result.getUserId());
+    }
+
+    @Test
+    void whenUserAndPostFoundThenReturnLike() {
+        long userId = 1L;
+        long postId = 2L;
+        UserEntity userEntity = mock(UserEntity.class);
+        PostEntity postEntity = mock(PostEntity.class);
+        LikeDto likeDto = mock(LikeDto.class);
+
+        when(likeDto.getUserId()).thenReturn(userId);
+        when(likeDto.getPostId()).thenReturn(postId);
+        when(userRepository.findById(userId)).thenReturn(Optional.of(userEntity));
+        when(postRepository.findById(postId)).thenReturn(Optional.of(postEntity));
+        when(userEntity.getId()).thenReturn(userId);
+        when(postEntity.getId()).thenReturn(postId);
+
+        LikeEntity result = likeMapper.toEntity(likeDto);
+        assertEquals(userEntity, result.getUserEntity());
+        assertEquals(postEntity, result.getPostEntity());
+    }
+
+    @Test
+    void whenUserNotFoundThenReturn404() {
+        long userId = 2L;
+        LikeDto likeDto = mock(LikeDto.class);
+
+        when(likeDto.getUserId()).thenReturn(userId);
+        when(userRepository.findById(userId)).thenReturn(Optional.empty());
+
+        ResponseStatusException exception =
+                assertThrows(ResponseStatusException.class, () -> likeMapper.toEntity(likeDto));
+        assertEquals("404 NOT_FOUND \"User 2 not found.\"", exception.getMessage());
+    }
+
+    @Test
+    void whenPostNotFoundThenReturn404() {
+        long postId = 2L;
+        long userId = 2L;
```

Comment:
bad test, If you would switch userId and postId in tested method test will still pass.